### ISSUE TITLE
Properly set IP Mode of Load Balancer Addresses to fix internal PROXY traffic

### ISSUE THREAD

**Issue by apricote** [hetznercloud/hcloud-cloud-controller-manager#727]
### TL;DR

[KEP-1860](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1860-kube-proxy-IP-node-binding) introduced a new field `service.status.loadBalancer.ingress[].ipMode: VIP | Proxy` to indicate the behavior of the Load Balancer.

Currently users on IPVS-based networking setups can not access the Load Balancer IPs when they enable PROXY protocol.

### Expected behavior

Users should always be able to access their services from inside of the cluster, even if they use IPVS and PROXY protocol.

IIUC we should set the IP Mode to `Proxy` for the IPs we return in the status.

**Comment by BartVB** [comment]
Would be fantastic if this could be implemented!

I have been banging my head against the wall for quite a bit of time when running into this issue. It results in rather strange/unexpected behaviour in the cluster. Would be a BIG help if this status field could be added to prevent others from wasting time when trying to trace this issue.

**Comment by lukasmetzner** [comment]
Added in #783
